app/controllers/users_controller.rb

```diff
@@ -72,4 +77,21 @@ def delete_admin
       redirect_to "/"
     end
   end
+
+  def google_oauth2
+    auth_hash = request.env["omniauth.auth"]
+    user = User.from_omniauth(auth_hash)
+    url = session[:return_to]|| root_path
```

Comment:
Useless assignment to variable - `url`.<br>Surrounding space missing for operator `||`.